apps/aechannel/src/aesc_client_reconnect_tx.erl

```diff
@@ -0,0 +1,193 @@
+    error(off_chain_tx).
+
+-spec signers(tx(), aec_trees:trees()) -> {ok, list(aec_keys:pubkey())}.
+signers(#channel_client_reconnect_tx{role = Role} = Tx, Trees) ->
+    case aec_chain:get_channel(channel_pubkey(Tx), Trees) of
+        {ok, Channel} ->
+            case Role of
+                initiator ->
+                    {ok, [aesc_channels:initiator_pubkey(Channel)]};
+                responder ->
+                    {ok, [aesc_channels:responder_pubkey(Channel)]}
+            end;
+        {error, not_found} -> {error, channel_does_not_exist}
+    end.
+
+-spec serialize(tx()) -> {vsn(), list()}.
+serialize(#channel_client_reconnect_tx{
+             channel_id = ChannelId
+           , role       = Role
+           , pub_key    = Pubkey }) ->
+    {?INITIAL_VSN,
+     [ {channel_id, ChannelId}
+     , {role      , atom_to_binary(Role, utf8)}
+     , {pub_key   , Pubkey}
+     ]}.
+
+-spec deserialize(vsn(), list()) -> tx().
+deserialize(?INITIAL_VSN, [ {channel_id, ChannelId}
+                          , {role      , Role0}
+                          , {pub_key   , Pubkey} ]) ->
+    channel = aeser_id:specialize_type(ChannelId),
+    account = aeser_id:specialize_type(Pubkey),
+    Role = case binary_to_existing_atom(Role0, utf8) of
+               initiator = R -> R;
+               responder = R -> R
+           end,
+    #channel_client_reconnect_tx{
+       channel_id = ChannelId
+     , role       = Role
+     , pub_key    = Pubkey }.
+
+-spec for_client(tx()) -> map().
+for_client(#channel_client_reconnect_tx{
+              channel_id = ChannelId
+            , role       = Role
+            , pub_key    = Pubkey }) ->
+    #{ <<"channel_id">> => aeser_api_encoder:encode(id_hash, ChannelId)
+     , <<"role">>       => atom_to_binary(Role, utf8)
+     , <<"pub_key">>    => aeser_api_encoder:encode(account, Pubkey) }.
+
+serialization_template(?INITIAL_VSN) ->
+    [ {channel_id, id}
+    , {role      , binary}
+    , {pub_key   , id} ].
+
+%%%===================================================================
+%%% Getters
+%%%===================================================================
+
+-spec channel_pubkey(tx()) -> aesc_channels:pubkey().
+channel_pubkey(#channel_client_reconnect_tx{channel_id = ChannelId}) ->
+    aeser_id:specialize(ChannelId, channel).
+
+-spec channel_id(tx()) -> aesc_channels:id().
+channel_id(#channel_client_reconnect_tx{channel_id = ChannelId}) ->
+    ChannelId.
+
+-spec role(tx()) -> initiator | responder.
+role(#channel_client_reconnect_tx{ role = Role }) ->
+    Role.
+
+-spec round(tx()) -> aesc_channels:round().
+round(#channel_client_reconnect_tx{}) ->
+    error(invalid).
+
+-spec state_hash(tx()) -> binary().
+state_hash(#channel_client_reconnect_tx{}) ->
+    error(invalid).
+
+-spec version(tx()) -> non_neg_integer().
+version(_) ->
+    ?INITIAL_VSN.
+
+-spec valid_at_protocol(aec_hard_forks:protocol_vsn(), tx()) -> boolean().
+valid_at_protocol(_Protocol, _Tx) ->
+    true.
```

Comment:
🤔

Reply:
I see `channel_client_reconnect_tx` has gas_limit 0 so not included in microblock candidate though I do not understand how this is prevented from entering mempool.

Reply:
```suggestion
    false.
```